qiskit_experiments/framework/composite/composite_analysis.py

```diff
@@ -120,242 +115,36 @@ def copy(self):
         ret._analyses = [analysis.copy() for analysis in ret._analyses]
         return ret
 
-    def run(
-        self,
-        experiment_data: ExperimentData,
-        replace_results: bool = False,
-        **options,
-    ) -> ExperimentData:
-        # Make a new copy of experiment data if not updating results
-        if not replace_results and _requires_copy(experiment_data):
-            experiment_data = experiment_data.copy()
-
-        if not self._flatten_results:
-            # Initialize child components if they are not initialized
-            # This only needs to be done if results are not being flattened
-            self._add_child_data(experiment_data)
-
-        # Run analysis with replace_results = True since we have already
-        # created the copy if it was required
-        return super().run(experiment_data, replace_results=True, **options)
-
     def _run_analysis(self, experiment_data: ExperimentData):
-        # Return list of experiment data containers for each component experiment
-        # containing the marginalized data from the composite experiment
-        component_expdata = self._component_experiment_data(experiment_data)
+        child_data = experiment_data.child_data()
 
-        # Run the component analysis on each component data
-        for i, sub_expdata in enumerate(component_expdata):
+        if len(self._analyses) != len(child_data):
+            # Child data is automatically created when composite result data is added.
+            # Validate that child data size matches with number of analysis entries.
+            experiment_data.create_child_data()
```

Comment:
```suggestion
        if len(child_data) == 0:
            # Child data is automatically created when composite result data is added.
            # Validate that child data size matches with number of analysis entries.
            experiment_data.create_child_data()
        if len(self._analyses) != len(child_data):
            raise("...")
```
I think this is more robust.